lib/matplotlib/tests/test_api.py

```diff
@@ -7,26 +11,33 @@
 from matplotlib import _api
 
 
+if typing.TYPE_CHECKING:
+    from typing_extensions import Self
```

Comment:
We have generally avoided typing_extensions (though admittedly I think it is already a transitive dependency anyway, so perhaps shouldn't be so worried about it)

`Self` is relatively new, so we have been largely just not type hinting things that would get `Self`, at least until it is supported in the stdlib for all supported python versions

In general, I _only_ added type hints to tests when mypy actually complained.

Reply:
I assumed it'd be okay as it's a dependency of `mypy` itself.

`Self` was mostly there to try to figure out `deprecate_privatize_attribute`, which might be impossible anyway as you've noted below.

Reply:
Same comment as in #26385, so can be handled in either PR: should at least acknowledge that we are directly using typing extensions, even if only in a dev dependency such as `requirements/testing/mypy.txt`, just to be safe.(setuptools-scm is our only hard dependency that depends on typing extensions, though `mypy` does as well, as you point out... but pyright does not depend on it, so if the setuptools-scm dependency gets removed could throw some users for a loop in minimal installs, but would have to be quite minimal)

Reply:
Handled in #26835, no need to double up/cause conflicts